src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContextHelpers.cs

```diff
@@ -36,6 +43,14 @@ internal static void VerifyArguments(Diagnostic diagnostic, Func<Diagnostic, boo
             {
                 throw new ArgumentException(string.Format(CodeAnalysisResources.UnsupportedDiagnosticReported, diagnostic.Id), nameof(diagnostic));
             }
+
+            if (string.IsNullOrEmpty(diagnostic.Id) || !diagnostic.Id.All(UnicodeCharacterUtilities.IsIdentifierPartCharacter))
```

Comment:
Can you please instead use the ISO definition of an identifier (e.g. doesn't start with a digit), and include `_` in the set of characters accepted?